hydromt_wflow/workflows/connect.py

```diff
@@ -0,0 +1,283 @@
+        raise ValueError(f"One or more variables missing from ds_model: {dvars_model}")
+
+    # Reproject
+    if gdf_riv.crs != ds_model.raster.crs:
+        gdf_riv = gdf_riv.to_crs(ds_model.raster.crs)
+    # Derive flwdir
+    flwdir = hydromt.flw.flwdir_from_da(ds_model["flwdir"])
+
+    # If tributaries or subbasins area method,
+    # need to derive the tributaries areas first
+    if connection_method == "subbasin_area" or add_tributaries:
+        logger.info("Linking 1D river to wflow river")
+        # 1. Derive the river edges / boundaries
+        # merge multilinestrings in gdf_riv to linestrings
+        riv1d = gdf_riv.explode().reset_index(drop=True)
+        # get the edges of the riv1d
+        riv1d_edges = riv1d.geometry.apply(lambda x: Point(x.coords[0]))
+        riv1d_edges = pd.concat(
+            [riv1d_edges, riv1d.geometry.apply(lambda x: Point(x.coords[-1]))]
+        )
+        # find geometry that are unique in riv1d_edges
+        riv1d_edges = gpd.GeoDataFrame(
+            riv1d_edges[~riv1d_edges.duplicated(keep=False)],
+            crs=riv1d.crs,
+            geometry="geometry",
+        )
+
+        # 2. snap edges to wflow river
+        # TODO if uparea column in riv1d, use it to snap to the closest river
+        # based on upstream area
+        da_edges, idxs, ids = hydromt.flw.gauge_map(
+            ds_model,
+            xy=(riv1d_edges.geometry.x, riv1d_edges.geometry.y),
+            stream=ds_model["rivmsk"].values,
+            flwdir=flwdir,
+            logger=logger,
+        )
+        points = gpd.points_from_xy(*ds_model.raster.idx_to_xy(idxs))
+        # if csv contains additional columns, these are also written in the staticgeoms
+        riv1d_edges = gpd.GeoDataFrame(
+            index=ids.astype(np.int32), geometry=points, crs=ds_model.raster.crs
+        )
+
+        # 3. Derive the subbasins corresponding to the river edges
+        da_edges_subbas, _ = hydromt.flw.basin_map(
+            ds_model, flwdir=flwdir, xy=(riv1d_edges.geometry.x, riv1d_edges.geometry.y)
+        )
+        da_edges_subbas.raster.set_crs(ds_model.raster.crs)
+        # convert to gdf
+        gdf_edges_subbas = da_edges_subbas.raster.vectorize()
+
+        # 4. Filter which subbasins are the upstream ones (tributaries)
+        # and which ones are the downstream ones (main river)
+        # and should be split into subbasins
+        # First intersect riv1d with gdf_edges_subbas
+        rivmerge = gpd.overlay(riv1d, gdf_edges_subbas).explode().reset_index(drop=True)
+        # Compute len of river
+        if rivmerge.crs.is_geographic:
+            rivmerge["len"] = rivmerge.geometry.to_crs(3857).length
+        else:
+            rivmerge["len"] = rivmerge.geometry.length
+        # Groupby value and sum length
+        rivmerge = rivmerge.groupby("value")["len"].sum()
+        # Select the subcatch where rivlength is more than 5times rivlen_avg
```

Comment:
Just a technical question: how sensitive are the results to this factor of 5?

Reply:
Good one! I think not too much. So you have the steps where I get the upstream and downstream boundaries of the 1D river and I need to snap these nodes to a wflow river cell before creating the subbasins. So depending on how far we had to snap, some parts of the river 1D lines are still included in the upstream subcatchment that I want to exclude.
So only if we had to snap too far then it may get sensitive. Let's see with practice if this is good enough or if it needs to be refined.

Reply:
Thanks for clarifying, makes sense!